src/platform-aws.c

```diff
@@ -580,7 +580,11 @@ int platform_init(const char **provider_filter)
 
 	domain_per_thread = ofi_nccl_domain_per_thread();
 	if (domain_per_thread == -1) {
-		domain_per_thread = platform_data->domain_per_thread;
+		if (platform_data) {
```

Comment:
I know there's remnants of Rashika in this code, but we really should explicitly check against NULL, rather than implicitly cast.

Reply:
Done